### PR TITLE
Drop Silverlight  & Windows Phone support. Xamarin support now only with .NET Standard. Removed loading for Xamarin assets

### LINKED ISSUE #2871: Drop Silverlight and WinPhone8 support

### PR DESCRIPTION
**Pull request by 304NotModified**
fixes #2871 

warning / open for discussion

- TryLoadFromAndroidAssets removed
- TLS_WORKAROUND removed
- What to do with NLog.Internal.Xamarin.PreserveAttribute? ()
- What to list in support list? Android? IOS? Xamarin?
- removed "// For issue #1351 - These are not available for Android or IOS. Is this correct?  -> https://github.com/NLog/NLog/pull/3978/files#diff-111027e98754fa8d87fe750d4edbd9f7L52-L56

FYI

- 736 `#if` in master
- 675 `#if` in dev
- now 500 `#if` in this branch

### CODE REVIEW

**Comment by 304NotModified** [comment]
maybe revert the format changes in src/NLog/Logger-V1Compat.cs

**Comment by snakefoot** [comment]
Guess when this has been merged, then no more big changes to master-branch.

**Comment by 304NotModified** [comment]
Indeed :)

**Comment by snakefoot** [comment]
> What to do with NLog.Internal.Xamarin.PreserveAttribute? 

Believe it should stay, since it is recognized by the Xamarin when compiling to native code.

**Comment by 304NotModified** [comment]
Before merging, I think these are important:

- **TryLoadFromAndroidAssets removed** - I'm not sure if this is OK, but any idea if this is possible for .NET standard? 
- **TLS_WORKAROUND removed** - I think this is fine
- **What to do with NLog.Internal.Xamarin.PreserveAttribute?** - so keep it
- **What to list in support list? Android? IOS? Xamarin?** - Think still in the support list
- **removed "// For issue #1351 - These are not available for Android or IOS. Is this correct?** -> https://github.com/NLog/NLog/pull/3978/files#diff-111027e98754fa8d87fe750d4edbd9f7L52-L56 - I'm not sure if this is an issue! 

@snakefoot please your opinion on those, thanks!

**Comment by snakefoot** [comment]
> TryLoadFromAndroidAssets removed - 

Think it is great that it is removed. Reducing the platform specific surprises is the focus for NetCore.

> TLS_WORKAROUND removed 

Think it is fine

> What to list in support list? Android? IOS? Xamarin?

Yes they are still supported automatically through NetStandard

> removed "// For issue #1351 - These are not available for Android or IOS. Is this correct? -

Was thinking mail-target should be moved output of NLog-core. And think it is fine that NetCore is not using app.config.

File: src/NLog/Internal/AssemblyHelpers.cs [hunk 1]
```diff
@@ -95,13 +86,6 @@ public static Assembly LoadFromName(string assemblyName)
 #if NETSTANDARD1_0 || WINDOWS_PHONE
```
[the remark this answers is not in the document]
Comment (304NotModified) [reply src/NLog/Internal/AssemblyHelpers.cs]: Thanks, that was the only one I missed :)

**Comment by 304NotModified** [comment]
All green expect publish to sonarcloud.

**Comment by snakefoot** [comment]
Searched the Wiki, and updated all pages that contained the word "Silverlight"

**Comment by snakefoot** [comment]
Searched the wiki, and updated all pages that contained the word "Compact Framework".

**Pull request merged**